File: swift/Main/03_leap/Sources/Leap/Leap.swift
//Solution goes in Sources

struct Year {

    // MARK: - Properties

    let calendarYear: Int
    let isLeapYear: Bool

    // MARK: - Init

    init(calendarYear: Int) {
        self.calendarYear = calendarYear

        let isDivisibleByFour = self.calendarYear % 4 == 0
        let isDivisibleByOneHundred = self.calendarYear % 100 == 0
        let isDivisilbeByFourHundred = self.calendarYear % 400 == 0

        self.isLeapYear = isDivisibleByFour && !isDivisibleByOneHundred || isDivisilbeByFourHundred
    }

}

File: typescript/Main/02_leap/leap.ts
function isLeapYear(year: number) {
    const isDivisibleByFour = year % 4 === 0
    const isDivisibleByOneHundred = year % 100 === 0
    const isDivisilbeByFourHundred = year % 400 === 0

    return isDivisibleByFour && !isDivisibleByOneHundred || isDivisilbeByFourHundred
}

export default isLeapYear